test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiDiscriminatorTests.cs

```diff
@@ -47,5 +51,51 @@ public void ParseBasicDiscriminatorShouldSucceed()
                     });
             }
         }
+
+        [Fact]
+        public void OneOfSchemasWithoutDiscriminatorPropertyShouldYieldError()
+        {
+            using (var stream = Resources.GetStream(Path.Combine(SampleFolderPath, "invalidOneOfDiscriminator.yaml")))
+            {
+                var diagnostic = new OpenApiDiagnostic();
+                var reader = new OpenApiStreamReader();
+                var doc = reader.Read(stream, out diagnostic);
+                diagnostic.Errors.ShouldAllBeEquivalentTo(new List<OpenApiError>
+                {
+                    new OpenApiValidatorError(nameof(OpenApiSchemaRules.ValidateOneOfDiscriminator),
+                        "#/paths/~1pets~1{id}/get/responses/200/content/application~1json/schema/oneOf",
+                        string.Format(SRResource.Validation_SchemaDoesntContainDiscriminatorPropertyInRequiredFieldList, "pet1", "petType")),
+
+                    new OpenApiValidatorError(nameof(OpenApiSchemaRules.ValidateOneOfDiscriminator),
+                        "#/paths/~1pets~1{id}/get/responses/200/content/application~1json/schema/oneOf",
+                        string.Format(SRResource.Validation_SchemaDoesntContainDiscriminatorProperty, "pet2", "petType")),
+
+                    new OpenApiValidatorError(nameof(OpenApiSchemaRules.ValidateOneOfDiscriminator), 
+                        "#/paths/~1pets~1{id}/get/responses/200/content/application~1json/schema/oneOf",
+                        string.Format(SRResource.Validation_SchemaDoesntContainDiscriminatorPropertyInRequiredFieldList, "pet2", "petType"))
+                });
+            }
+        }
+
+        [Fact]
+        public void AnyOfSchemasWithoutDiscriminatorPropertyShouldYieldError()
+        {
+            using (var stream = Resources.GetStream(Path.Combine(SampleFolderPath, "invalidAnyOfDiscriminator.yaml")))
+            {
+                var diagnostic = new OpenApiDiagnostic();
+                var reader = new OpenApiStreamReader();
+                var doc = reader.Read(stream, out diagnostic);
+                diagnostic.Errors.ShouldAllBeEquivalentTo(new List<OpenApiError>
+                {
+                    new OpenApiValidatorError(nameof(OpenApiSchemaRules.ValidateAnyOfDiscriminator),
+                        "#/paths/~1pets~1{id}/get/responses/200/content/application~1json/schema/anyOf",
+                        string.Format(SRResource.Validation_SchemaDoesntContainDiscriminatorProperty, "pet2", "petType")),
+
+                    new OpenApiValidatorError(nameof(OpenApiSchemaRules.ValidateAnyOfDiscriminator), 
+                        "#/paths/~1pets~1{id}/get/responses/200/content/application~1json/schema/anyOf",
+                        string.Format(SRResource.Validation_SchemaDoesntContainDiscriminatorPropertyInRequiredFieldList, "pet2", "petType"))
+                });
+            }
+        }
```

Comment:
These tests belong to Microsoft.OpenApi.Tests/Validations. Could you move them there and adjust?